org.eclipse.xtext.java.tests/src/org/eclipse/xtext/java/tests/JavaSourceLanguageTest.java

```diff
@@ -217,9 +213,19 @@ public void testJava21Record() {
 
 	@Inject
 	private IJvmTypeProvider.Factory typeProviderFactory;
-
+	
 	protected XtextResourceSet resourceSet(Map<String, String> files) {
+		return this.resourceSet(files, null);
+	}
+
+	protected XtextResourceSet resourceSet(Map<String, String> files, JavaVersion javaVersion) {
 		XtextResourceSet result = resourceSetProvider.get();
+		if (javaVersion != null) {
+			JavaConfig javaConfig = new JavaConfig();
+			javaConfig.setJavaSourceLevel(JavaVersion.JAVA21);
```

Comment:
I guess here you meant to pass `javaVersion`, the parameter, not the hardcoded version

Reply:
yes